Appc 1229 remove handshake from the poa star

brunovarelaAptoide opened the pull request:
**What does this PR do?**

This PR removes the hand shake of POA and now activates directly the WalletPoaService.
This PR also removes the bug of the POA duration be only 1 min and 50 secs and now its the real duration of 2 minutes.
Removed Tray dependency

**Database changed?**

No

**Where should the reviewer start?**

-PoAManager.java
-PoAServiceConnectorImpl.java

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APPC-1229](https://aptoide.atlassian.net/browse/APPC-1229)

This should be tested with the Appcoins dev wallet and with the toolbox in Debug mode.
Also the No wallet install flow and PoA starting after install wallet should be tested.

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1229](https://aptoide.atlassian.net/browse/APPC-1229)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 
No



**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

brunovarelaAptoide committed:
changes

- changes to remove handshake

brunovarelaAptoide committed:
Revert "changes"

This reverts commit 4b0fd2ff393b00d782504f2daf0b7c130c8f84e4.

brunovarelaAptoide committed:
Merge remote-tracking branch 'remotes/origin/dev' into APPC-1229-Remove-handshake-from-the-PoA-star

brunovarelaAptoide committed:
Merge remote-tracking branch 'remotes/origin/dev' into APPC-1229-Remove-handshake-from-the-PoA-star

brunovarelaAptoide committed:
remove handshake working

- Version of the remove handshake working

brunovarelaAptoide committed:
Changes

Changed build gradle properties

brunovarelaAptoide committed:
changes

-Changed build gradle

brunovarelaAptoide committed:
BugFix

- Poa time was 1 min and it was fixed to 2 min.

brunovarelaAptoide committed:
Bug fix

- POA bug fix: the poa duration time ws 1min and 50 and it was suppose to be 2 min. Fixed

brunovarelaAptoide committed:
Changes

Removed tray dependency

mikieribeiro commented:
Poa working for each case but sms validation is returning an "INVALID_PHONE" error (probably not related to this PR but just leaving it here)

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java:
```diff
@@ -203,11 +203,12 @@ private void sendProof() {
+          Log.e(TAG, "Start proof sending");
+          postponeSendProof();
+          firstProof = true;
+          return;
```
Don't use the return to stop the process when you are in the middle of several conditions that are not using return statement to finish it, just use if else to do it.

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java:
```diff
@@ -203,11 +203,12 @@ private void sendProof() {
-        // the connection is done.
-        poaConnector.connectToService(appContext);
-        // send proof
+        if (!firstProof) {
```
Don't create a variable to control the first proof sent when we already have a variable that controls the number of proofs sent, in this case we already have the 'proofsSent' variable to control the number of proofs sent.

JoaoRaimundo replied:
Maybe use the logic to postpone the proof to be sent on the initiateProofSending method instead of doing it in the sendProof method

brunovarelaAptoide replied:
Removed it and changed to initiateProofSending.

brunovarelaAptoide committed:
Changes Code Review

- Removed control variable to avoid POA to be called 10 secs later.
- Changed initProofSending to start POA 10 secs later

JoaoRaimundo commented:
Please also remove references to 'grandcentrix' and 'ver.tray'

brunovarelaAptoide committed:
changes codereview

- Removed references to 'grandcentrix' and 'ver.tray'

brunovarelaAptoide committed:
Code review changes

- Removed unused variable
